ledger/src/blockstore.rs

```diff
@@ -349,6 +444,74 @@ impl BlockstoreInsertionMetrics {
     }
 }
 
+impl BlockstoreRocksDbColumnFamilyMetrics {
+    pub fn report_metrics(&self, cf_name: &'static str) {
+        datapoint_info!(
+            "blockstore_rocksdb_cf_metrics",
+            ("cf_name", cf_name, String),
```

Comment:
Because datapoint_info's metric name must be a static string, I created a field for column family name.  So in the dashboard, we might want to group by `cf_name`.

Reply:
I think you can drop the "metrics" and change "blockstore_rocksdb_cf_metrics" to "blockstore_rocksdb_cfs"; we don't have a similar suffix on any other metrics.

I'm not sure if this will work, but what about passing the metric name into `report_metrics()` as an argument, and then declaring those strings in `submit_rocksdb_cf_metrics_for_all_cfs()`; something like:
```
    pub fn submit_rocksdb_cf_metrics_for_all_cfs(&self) {
        self.submit_rocksdb_cf_metrics::<cf::SlotMeta>("blockstore_rocksdb-meta_cf");
        ...
```

Reply:
> I think you can drop the "metrics" and change "blockstore_rocksdb_cf_metrics" to "blockstore_rocksdb_cfs"; we don't have a similar suffix on any other metrics.

Sounds good to me.

> I'm not sure if this will work, but what about passing the metric name into report_metrics() as an argument, and then declaring those strings in submit_rocksdb_cf_metrics_for_all_cfs(); something like

That is my original thinking, but I am currently feeling that having cf_name as a field might be a good idea as we are more flexible on the query.  Although I need to learn more about Chronograf. Some syntaxes are non-intuitive xD.

Reply:
>but I am currently feeling that having cf_name as a field might be a good idea as we are more flexible on the query.

Not sure I follow here.

In any case, I think it is fair to say that we want to be able to view graphs. The following query will display a graph:
```
SELECT "total_sst_files_size" FROM "mainnet-beta"."autogen"."blockstore_rocksdb_cf_metrics"
WHERE "cf_name"='code_shred' AND time > :dashboardTime: AND time < :upperDashboardTime: AND "host_id"='4Porfzn25m8HDeaJwf9amwnAPMRDHoiVb5HBKC477Tuh'
```
**Note that in the `WHERE` clause, using ='code_shred' (single quote) works whereas ="code_shred" (double quote) does not.
<img width="1280" alt="image" src="https://user-images.githubusercontent.com/5400107/156666967-5a86732a-d02a-45f8-aba6-a8033d8c81da.png">

This shows me reasonable data for one column; however, we are unable to group by `cf_name`. If I expand the `WHERE` clause to accept a regex (such as `WHERE "cf_name" =~ /code_shred|data_shred/`), I get the data for all 2 in one trace (which causes a sawtooth graph). In order to group by `cf_name`, we'd need to pull it in in the `SELECT` clause. In doing so, we start pulling in string data which makes us unable to view a graph. So, this is a bit of a catch-22 situation.

Lumping each column under its' own metric name would support the case of just finding the metric and instantly having the graph ..

Reply:
Not being able to group by "cf_name" is really a pain.  Hopefully, there is a way but we just have not yet figured it out.
Having them in one table might allow us to easily do a sum across different CFs to get a bigger picture.

Let me play around a bit to see whether `group by` is something possible.

Reply:
Okay, sad news about `group by`:

https://docs.influxdata.com/influxdb/v1.7/query_language/data_exploration/

```
Note: You cannot use GROUP BY to group fields.
```

But from the discussion [here](https://stackoverflow.com/questions/51793362/influxdb-group-by-field-not-tag-and-get-top-10), looks like it is something doable if I insert cf_name as a tag instead of a field will work.  Let me try it out a little bit.